### PR TITLE
Make ipc-channel an optional feature.

### LINKED ISSUE #558: Support using WebRender without IPC
It looks like we're going to be using Gecko IPC with WebRender instead of the Servo IPC. In this situation we don't want to be eating the overhead of an additional IPC serialization/copy.

### PR DESCRIPTION
**Pull request by glennw**
This allows clients to opt-in to IPC for communication if required.

Otherwise, in-process rust channels are used, avoiding serialization.

This fixes #558.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/572)

### CODE REVIEW

**Comment by glennw** [comment]
cc @jrmuizel

File: webrender_traits/src/channel.rs [hunk 1]
```diff
@@ -0,0 +1,148 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use serde::{Deserialize, Serialize};
+use std::io::Error;
+
+#[cfg(feature = "ipc")]
```
Comment (kvark) [inline webrender_traits/src/channel.rs]: given that everything in this module is split between `ipc` and non-ipc, I'd rather see separate files `channel.rs` and `channel_ipc.rs` with a single switch upon including them.

Comment (glennw) [inline webrender_traits/src/channel.rs]: Good idea - https://github.com/servo/webrender/issues/573

**Pull request merged**